extensions/amp-list/0.1/amp-list.js

```diff
@@ -924,18 +921,19 @@ export class AmpList extends AMP.BaseElement {
           // Necessary since load-more elements are toggled in the above block
           this.attemptToFitLoadMore_(dev().assertElement(this.container_));
         }).catch(() => {
-          this.mutateElement(() =>
-            this.getLoadMoreService_().setLoadMoreFailed())
-              .then(() => {
-                this.attemptToFitLoadMoreElement_(
-                    this.getLoadMoreService_().getLoadMoreFailedElement(),
-                    dev().assertElement(this.container_));
-                const loadMoreFailedClickable = this.getLoadMoreService_()
-                    .getLoadMoreFailedClickable();
-                this.unlistenLoadMore_ = listen(
-                    loadMoreFailedClickable,
-                    'click', () => this.loadMoreCallback_(/*opt_reload*/ true));
-              });
+          this.handleLoadMoreFailed_();
```

Comment:
Turns out that taking the error throw out of `fetchList` is more involved and requires a bit more moving things around. Basically `maybeSetLoadMore` is a function that should only be called once after the fetch is done (in layoutCallback and mutatedAttributesCallback) to check for whether we should show the load-more-button or any relevant load-more elements. But better test coverage is necessary for me to be comfortable with that change.